src/helper/exception.hpp

```diff
@@ -99,4 +99,12 @@ class client_disconnect : public std::exception {
     }
 };
 
+class unexpected_command : public std::exception {
```

Comment:
You could add the command name as a parameter to the constructor to improve the error message.